tests/integration_tests/data_steward/cdr_cleaner/cleaning_rules/clean_smoking_test.py

```diff
@@ -0,0 +1,111 @@
+        observation_id,
+        person_id,
+        observation_concept_id, 
+        observation_source_concept_id,
+        value_as_concept_id,
+        value_source_concept_id,
+        observation_date,
+        observation_type_concept_id
+    )
+    VALUES
+        (1, 11, 1, 1585866, 45877994, 1, date('2000-01-01'), 0),
+        (2, 12, 2, 903062, 45877994, 2, date('2000-01-01'), 0),
+        (3, 13, 40770349, 1585873, 1177221, 903079, date('2000-01-01'), 0),
+        (4, 14, 4, 1585865, 45877994, 4, date('2000-01-01'), 0),
+        (5, 15, 5, 1585871, 45877994, 5, date('2000-01-01'), 0),
+        (6, 16, 40766333, 1585864, NULL, NULL, date('2000-01-01'), 0),
+        (7, 17, 7, 1585864, 903096, 7, date('2000-01-01'), 0),
+        (8, 18, 8, 1585870, 903096, 8, date('2000-01-01'), 0),
+        (9, 19, 40770349, 1585873, 903096, 903096, date('2000-01-01'), 0),
+        (10, 20, 40766333, 1585864, 45876636, 903087, date('2000-01-01'), 0)
+""")
+
+
+class CleanSmokingPpiTest(BaseTest.CleaningRulesTestBase):
+
+    @classmethod
+    def setUpClass(cls):
+        print('**************************************************************')
+        print(cls.__name__)
+        print('**************************************************************')
+
+        super().initialize_class_vars()
+
+        cls.project_id = os.environ.get(PROJECT_ID)
+        cls.dataset_id = os.environ.get('RDR_DATASET_ID')
+        cls.sandbox_id = f'{cls.dataset_id}_sandbox'
+        cls.lookup_table = SMOKING_LOOKUP_TABLE
+
+        cls.rule_instance = CleanSmokingPpi(cls.project_id, cls.dataset_id,
+                                            cls.sandbox_id)
+
+        cls.fq_table_names = [
+            f'{cls.project_id}.{cls.dataset_id}.{OBSERVATION}'
+        ]
+
+        cls.fq_sandbox_table_names = [
+            f'{cls.project_id}.{cls.sandbox_id}.{cls.rule_instance.sandbox_table_for(OBSERVATION)}',
+            f'{cls.project_id}.{cls.sandbox_id}.{cls.lookup_table}'
+        ]
+
+        super().setUpClass()
+
+    def setUp(self):
+
+        super().setUp()
+
+        self.load_test_data([
+            LOAD_QUERY.render(project_id=self.project_id,
+                              dataset_id=self.dataset_id)
+        ])
+
+    def test_setup_rule(self):
+        # run setup_rule and see if the affected_table is updated
+        self.rule_instance.setup_rule(self.client)
+
+        # sees that setup worked and reset affected_tables as expected
+        self.assertEqual(set(OBSERVATION),
+                         set(self.rule_instance.affected_tables))
+
+    def test_clean_smoking_ppi(self):
+        """
+        Tests that the specifications for the queries perform as designed.
+        """
+
+        tables_and_counts = [{
+            'fq_table_name':
+                f'{self.project_id}.{self.dataset_id}.{OBSERVATION}',
+            'fq_sandbox_table_name':
+                self.fq_sandbox_table_names[0],
```

Comment:
As of now, the rule is not sandboxing the original data.  It is instead creating a table with the corrected data in the sandbox.  You can either:
1. Point this test line at the `NEW_SMOKING_ROWS` table.
2. OR Start sandboxing the data before it is deleted.  Then point the test line at the sandboxed original data.
3. OR Do both 1 and 2.
The first item is the smallest change to the existing code.  It has no impacts on the generated SQL. The implementation of the second item will introduce a little more SQL and change the test code.  The third implementation is the most exhaustive, but may be harder to implement the test.  Implementing the first item is the fastest and most important fix.  The other two options are "nice to haves".

Reply:
At this point, I'll stick with the first method so as to not confuse myself more.